src/main/java/kitchenpos/products/tobe/application/ProductService.java

```diff
@@ -0,0 +1,64 @@
+package kitchenpos.products.tobe.application;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import kitchenpos.menus.domain.Menu;
+import kitchenpos.menus.domain.MenuProduct;
+import kitchenpos.menus.domain.MenuRepository;
+import kitchenpos.products.tobe.domain.Product;
+import kitchenpos.products.tobe.infra.ProductRepository;
+import kitchenpos.products.tobe.infra.PurgomalumClient;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+    private final MenuRepository menuRepository;
+    private final PurgomalumClient purgomalumClient;
+
+    public ProductService(
+        final ProductRepository productRepository,
+        final MenuRepository menuRepository,
+        final PurgomalumClient purgomalumClient
+    ) {
+        this.productRepository = productRepository;
+        this.menuRepository = menuRepository;
+        this.purgomalumClient = purgomalumClient;
+    }
+
+    @Transactional
+    public ProductCreationResponseDto create(final String name, final BigDecimal price) {
+        final Product product = productRepository.save(Product.create(name, price, purgomalumClient));
+        return ProductCreationResponseDto.of(product);
+    }
+
+    @Transactional
+    public Product changePrice(final UUID productId, final BigDecimal price) {
+        final Product product = productRepository.findById(productId)
+            .orElseThrow(NoSuchElementException::new);
+        product.changePrice(price);
+        final List<Menu> menus = menuRepository.findAllByProductId(productId);
+        for (final Menu menu : menus) {
+            BigDecimal sum = BigDecimal.ZERO;
+            for (final MenuProduct menuProduct : menu.getMenuProducts()) {
+                sum = sum.add(
+                    menuProduct.getProduct()
+                        .getPrice()
+                        .multiply(BigDecimal.valueOf(menuProduct.getQuantity()))
+                );
+            }
+            if (menu.getPrice().compareTo(sum) > 0) {
+                menu.setDisplayed(false);
+            }
+        }
+        return product;
+    }
+
+    @Transactional(readOnly = true)
+    public List<Product> findAll() {
```

Comment:
여기도 도메인 모델을 외부로 노출하고 있네요? DTO를 적용해 보시는 게 좋을 것 같습니다.
```suggestion
    public List<ProductResponse> findAll() {
```

Reply:
@HackSung 
궁금한 부분이 있습니다
저는 실제 업무할 때 ProductResponse 라는건 포괄적인 느낌이 커서
클라이언트에게 응답데이터를 내려줄 때, '어떤 응답 데이터가 들어있어' 라고 의미를 내포해서 전달하고 있는데
학성님은 ResponseDto 내려줄 때 클래스 명을 어떻게 사용하고 계실까요?

Reply:
네 저도 제웅님 의견에 동의합니다!
네이밍을 할 때 가장 중요한 부분은 의도를 명확하게 나타낼 수 있도록 가능한 구체적인 이름을 붙이는 것이라고 생각합니다.
질문 주신 DTO 클래스명은 
저의 경우 먼저 팀원간 합의된 컨벤션을 지켜 일관된 형식의 이름을 사용하고 있습니다.
프로젝트에서 사용하는 DTO 이름의 구성 형식은 다음과 같습니다.

> `처리대상` + `조작행위` + `[복수]` + `[조건]` + `요청/응답구분`

위 형식에 따라 아래와 같은 DTO클래스명이 도출 되게 됩니다.
- 요청 DTO
  -  상품 단건 조회 : ProductSearchRequest
  -  상품 복수 조회 : ProductSearchAllRequest
  - 상품 저장 : ProductSaveRequest
  - 상품 수정 : ProductUpdateRequest
  - 상품 삭제 : ProductDeleteRequest
- 응답 DTO
  - 상품 단건 조회 : ProductSearchResponse
  - 상품 복수 조회 : ProductSearchAllResponse 
  - 상품 조건 조회 : ProductSearchAllByNameResponse
  - 상품 저장 : ProductSaveResponse
  - 상품 수정 : ProductUpdateResponse
  - 상품 삭제 : ProductDeleteResponse
 
참고가 되셨으면 합니다. 😄